html/db.php

```diff
@@ -0,0 +1,268 @@
+<?php
+// dbに接続する
+class Db
+{
+    private static $pdo = null;
+    public static function getPdo()
+    {
+        if (self::$pdo === null) {
+            try {
+                self::$pdo = new PDO('mysql:dbname=sns_api;host=mysql;charset=utf8', 'root', 'root');
+            } catch (PDOException $e) {
+                echo json_encode('DB接続エラー:' . $e->getMessage());
+                die();
+            }
+        }
+        return self::$pdo;
+    }
+
+    // sqlのprepareとbindValueを実行する関数
+    public static function prepareAndBindValue(string $sql, array $params = [])
+    {
+        $stmt = Db::getPdo()->prepare($sql);
+        if (!empty($params)) {
+            // パラメタがある
+            foreach ($params as $key => $ary) {
+                $value = $ary[0];
+                $type = $ary[1];
+                $stmt->bindValue($key, $value, $type);
+            }
+        }
+        return $stmt;
+    }
+
+    // sqlのexecuteを実行し結果を返す関数
+    public static function execute(string $sql, array $params = [])
+    {
+        $stmt = self::prepareAndBindValue($sql, $params);
+        return $stmt->execute();
+    }
+
+    // sqlのexecuteとfetchAllを実行しfetchAllの結果を返す関数
+    public static function executeAndFetchAll(string $sql, array $params = [])
+    {
+        $stmt = self::prepareAndBindValue($sql, $params);
+        $stmt->execute();
+        return $stmt->fetchAll(PDO::FETCH_ASSOC);
+    }
+
+
+
+
+
+    // selectUser系，特定のユーザーを取得するもの
+    // emailに合致するユーザを取得する
+    public static function selectUserByEmailFetchAll(string $email)
+    {
+        $sql = 'SELECT * FROM users WHERE email = :email';
+        $params = [
+            ':email' => [$email, PDO::PARAM_STR],
+        ];
+        return Db::executeAndFetchAll($sql, $params);
+    }
+
+    // tokenに合致するユーザーを取得する
+    public static function selectUserByTokenFetchAll(string $token)
+    {
+        $sql = 'SELECT * FROM users WHERE token = :token';
+        $params = [
+            ':token' => [$token, PDO::PARAM_STR],
+        ];
+        return Db::executeAndFetchAll($sql, $params);
+    }
+
+    // insertされたuserのidに合致するuserを再度取得する
+    public static function selectUserByInsertedUserIdFetchAll(int $insertedUserId)
+    {
+        $sql = 'SELECT id, name, bio, created_at, updated_at FROM users WHERE id = :id';
+        $params = [
+            'id' => [$insertedUserId, PDO::PARAM_INT],
```

Comment:
```suggestion
            ':id' => [$insertedUserId, PDO::PARAM_INT],
```
https://www.php.net/manual/ja/sqlite3stmt.bindvalue.php
名前にコロンを付け忘れるとPHPが勝手に補うのですね。知らなかった。
統一したいので他と同様コロンを付けましょう。